### PR TITLE
created the databse for trust registry 

### PR DESCRIPTION
**Pull request by vsvishalsharma**
### Summary
This pull request includes:
- Database design for the Trust Registry system.
- Implementation of POST /registries endpoint to create new registries in the Governance Authority table.

### Details
- **Database Design**: Designed PostgreSQL schema using Prisma ORM.
### GovernanceAuthority
- Acts as the root of trust in the system.
- Manages top-level authorities overseeing the trust framework.

### Organization
- Represents entities like issuers and verifiers within the trust framework.
- Links organizations to their governing authority, establishing trust chains.

### Schema
- Defines the structure of credentials that can be issued.
- Enables standardization of credential formats across the system.

### CredentialDefinition 
- Stores essential cryptographic parameters for credential issuance and verification. Usage varies as per needs; some implementations might require storing this data, while others might not.
- Key parameters include:
  - **Public Key**: For verifying credential signatures.
  - **Signature Type**: Indicates the cryptographic scheme used.
  - **Revocation Support**: Specifies whether this credential type can be revoked.
  - **Zero-Knowledge Proof Parameters**: For enabling privacy-preserving verifications.

### Credential
- Tracks issued credentials without storing sensitive data.
- Links credentials to their schema and definition.

### RevocationRegistry
- Manages the revocation status of issued credentials.
- Crucial for maintaining the current validity of credentials.

### Namespace
- Organizes trust frameworks into logical groupings.
- Facilitates management of large-scale trust ecosystems.

### AssuranceLevel
- Defines different levels of trust within the framework.
- Enables nuanced trust decisions based on predefined criteria.

### RegistryRelationship
- Establishes connections between different governance authorities.
- Enables implementation of "registry of registries" concept.

- **POST /registries Endpoint**: Added route, controller, service, and Prisma integration for registry creation (Governance authority as the owner of trust registry).
- Tested the route via postman

### Next Steps
- Implement additional endpoints for namespaces, assurance levels, organizations, schemas, and credentials.
- Perform testing and validation of implemented features.

### Additional Notes
- Planning for continuous integration and unit testing.

##Relates to #3

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+model Schema {
+  id              String       @id @default(uuid())
+  name            String
+  definition      Json
+  organizationId  String
+  organization    Organization @relation(fields: [organizationId], references: [id])
+  credentials     Credential[]
+}
+
+model Credential {
```
Comment (ankita-p17) [inline prisma/schema.prisma]: This can be removed, as credential details are not stored anywhere

File: prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  definitionId          String
+  definition            CredentialDefinition @relation(fields: [definitionId], references: [id])
+  revocationRegistryId  String?
+  revocationRegistry    RevocationRegistry?  @relation(fields: [revocationRegistryId], references: [id])
+}
+
+model CredentialDefinition {
+  id          String       @id @default(uuid())
+  name        String
+  definition  Json
```
Comment (ankita-p17) [inline prisma/schema.prisma]: definition can be renamed to definitionId with datatype String

File: prisma/schema.prisma (hunk 1 above)
Comment (ankita-p17) [reply prisma/schema.prisma]: All the reference to other table should also be removed

**Review by ankita-p17 — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 2 above)
Comment (vsvishalsharma) [reply prisma/schema.prisma]: - The CredentialDefinition model has also been removed. Instead, we've incorporated the necessary information directly into the Schema model.
- The Schema model now includes fields for both W3C and AnonCreds schemas. Here's how it looks in the current schema:
```
model Schema {
  id              String       @id @default(uuid())
  name            String
  type            SchemaType
  w3cUri          String?
  anonCredsDefinitionId String?
  organizationId  String
  organization    Organization @relation(fields: [organizationId], references: [id])
}

enum SchemaType {
  W3C
  ANONCREDS
}
```

File: prisma/schema.prisma (hunk 1 above)
Comment (vsvishalsharma) [reply prisma/schema.prisma]: Removed the Credential model and all its references from the schema.

**Review by ankita-p17 — APPROVED** [review]

**Comment by ankita-p17** [comment]
Looks good to merge

**Review by amitpadmani-awts — APPROVED** [review]
LGTM

**Pull request merged**